packages/storybook/src/stories/grid-rendering.stories.tsx

```diff
@@ -40,3 +52,33 @@ export const RenderInputInCell = () => {
     </div>
   );
 };
+export const InfiniteLoading = () => {
+  const apiRef = useGridApiRef();
+  const [size, setSize] = React.useState(50);
+  const { loading, data, setRowLength, loadNewData } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: size,
+    maxColumns: 20,
+  });
+
+  const handleOnScrollBottom = () => {
+    const newRowLength = size + 20;
+    setSize(newRowLength);
+    setRowLength(newRowLength);
```

Comment:
This `setRowLength` seems an anti-pattern. It seems much simpler to make the `rowLength` option of the hook controlled, and store the state outside anytime its needed.

Reply:
Ok, yes this is just to showcase that the solution is working but I was thinking of having more or less the same example for the docs example.

Reply:
> removing the setRowLength and only having the rowLength controlled […] but the scroll doesn't jump

Yes, it's my point and the problem I propose we fix by removing `setRowLength`. There is no controllable pattern. The state has to be kept in sync between two places.

Reply:
Ok, I've updated it. If this ok I'll go ahead and refactor the existing demos and stories that use `setRowLength` to no longer be dependant on it and remove the need to maintain the rowLenght state inside the `useDemoData` hook.

Reply:
@DanailH I personally think that it would be better. It's outside of the scope of fixing the infinite loading, but I wanted to raise it as we are depending on it.

Reply:
I'll open a separate PR with that change only because it seems to spark a discussion in itself.